src/config.coffee

```diff
@@ -665,12 +665,23 @@ class Config
   #
   # Returns an {Object} eg. `{type: 'integer', default: 23, minimum: 1}`.
   # Returns `null` when the keyPath has no schema specified.
+  # Returns `false` when the key-path is not accessible from the root schema.
   getSchema: (keyPath) ->
     keys = splitKeyPath(keyPath)
     schema = @schema
     for key in keys
-      break unless schema?
-      schema = schema.properties?[key]
+      if schema.type is 'object'
+        childSchema = schema.properties?[key]
+        unless childSchema?
+          if isPlainObject(schema.additionalProperties)
+            childSchema = schema.additionalProperties
+          else if schema.additionalProperties is false
+            return false
+          else
+            return null
```

Comment:
I am not 100% familiar with this code but I am wondering if we could avoid `null` and/or `false` as this function's return values. Without looking at the documentation, the calling code seems a bit obscure to me.

I am probably ok with keeping `null`, as it sounds reasonable that a schema cannot be found. But can we encode `false` differently? Can we tell what that value represents without indicating it in the documentation?

Maybe we could make the return value consistent across the various `return` statements, and return an object that always responds to the following messages:

* `hasSchemaSpecified`
* `isAccessibleFromRootSchema`

@maxbrunsfeld: what do you think?